desktop/libs/notebook/src/notebook/connectors/trino.py

```diff
@@ -274,11 +270,17 @@ def close_session(self, session):
 
 
   def _show_databases(self):
+    catalogs = self._show_catalogs()
+    hostname, port = self.get_hostname_and_port(self.api_url)
+    databases = []
 
-    query_client = TrinoQuery(self.db, 'SHOW SCHEMAS')
-    response = query_client.execute()
-    res = response.rows
-    databases = [item for sublist in res for item in sublist]
+    for catalog in catalogs:
+      trino_session = ClientSession(self.user.username, catalog)
+      trino_request = TrinoRequest(hostname, port, trino_session)
```

Comment:
Question: Not possible to re-use same session?

Reply:
No, via `SET SESSION` cmd we can only set the properties of a session or catalog properties not the name.